<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>DOG MICROBIOME BASIC ANALYSES</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>



<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<h1>DOG MICROBIOME BASIC ANALYSES</h1>

<h2>Background</h2>

<p>There are three dogs which are treated with increased percentage of a
compound in their diet: 5 different treatments (0-4, representing an
increased percentage of a compound in their diet) Analyses included
here:</p>

<ul>
<li>  Import .biom and .tre files generated in QIIME as well as metadata
file: merge these objects in a phyloseq object</li>
<li>  Basic data filter: assess number of reads/sample, plot rarefaction
curves, and exclude low abundance OTUs and samples that do not meet
minimum number of reads cutoff.</li>
<li>  Basic exploratory plots including bar plots, alpha- and
beta-diversity, heatmaps.</li>
<li>  Differential abundance testing by a) Dog and b) Treatment</li>
</ul>

<h2>This tutorial should be run in an interactive session. Please do not runnning anything on the headnode.</h2>

<h3>To get to a compute node do</h3>

<pre><code class="bash">qsub -I -q UCTlong -l walltime=08:00:00
</code></pre>

<p>Once you are on a compute node you will see that the prompt changes from <code>@srvslshpc001</code> to <code>@srvslshpc60X</code> e.g.</p>

<pre><code class="bash">gerrit@srvslshpc001:~&gt; qsub -I -q UCTlong -l walltime=08:00:00
qsub: waiting for job 1598565.srvslshpc001 to start
qsub: job 1598565.srvslshpc001 ready

gerrit@srvslshpc601:~&gt; hostname
srvslshpc601

</code></pre>

<h2>Make an output directory for R downstream analyses</h2>

<p><strong>NB - replace hpc30 with your user account</strong></p>

<pre><code class="r">mkdir /researchdata/fhgfs/hpc30/R_downstream
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;mkdir&#39; not found
</code></pre>

<h2>Load the R environment required</h2>

<pre><code>module load software/R-3.3.0
</code></pre>

<h2>Startup R </h2>

<pre><code>enter &#39;R&#39; into terminal
</code></pre>

<h2>Import data and create phyloseq object</h2>

<p><strong>Import BIOM file (generated in QIIME) into a phyloseq object</strong></p>

<pre><code class="r">library(phyloseq)
</code></pre>

<pre><code>## Loading required package: ade4
## Loading required package: picante
## Loading required package: ape
## Loading required package: vegan
## Loading required package: permute
## Loading required package: lattice
## This is vegan 2.4-2
## 
## Attaching package: &#39;vegan&#39;
## 
## The following object is masked from &#39;package:ade4&#39;:
## 
##     cca
## 
## Loading required package: nlme
</code></pre>

<pre><code class="r">library(ggplot2)
library(gridExtra)
library(dunn.test)
</code></pre>

<pre><code>## Error in library(dunn.test): there is no package called &#39;dunn.test&#39;
</code></pre>

<pre><code class="r">library(vegan)
library(randomForest)
</code></pre>

<pre><code>## randomForest 4.6-12
## Type rfNews() to see new features/changes/bug fixes.
## 
## Attaching package: &#39;randomForest&#39;
## 
## The following object is masked from &#39;package:gridExtra&#39;:
## 
##     combine
</code></pre>

<pre><code class="r">library(dplyr)
</code></pre>

<pre><code>## Error in library(dplyr): there is no package called &#39;dplyr&#39;
</code></pre>

<p><strong>Import custom functions used in script</strong></p>

<pre><code class="r">source(&quot;/scratch/DB/bio/training/16SrRNA/16SrRNA-hex-tutorial/src/microbiome_custom_functions.R&quot;)
</code></pre>

<pre><code>## Warning in grepl(&quot;\n&quot;, lines, fixed = TRUE): input string 545 is invalid in
## this locale
</code></pre>

<pre><code>## Loading required package: pkgmaker
## Loading required package: registry
## Loading required package: rngtools
## Loading required package: cluster
## NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 63/64
##   To enable shared memory capabilities, try: install.extras(&#39;
## NMF
## &#39;)
## 
## Attaching package: &#39;NMF&#39;
## 
## The following object is masked from &#39;package:nlme&#39;:
## 
##     coef&lt;-
## 
## The following object is masked from &#39;package:ape&#39;:
## 
##     consensus
</code></pre>

<pre><code>## Error in library(matrixStats): there is no package called &#39;matrixStats&#39;
</code></pre>

<p><strong>Set the working directory and import data</strong>
<strong>NB replace hpc30 with the name that has been given to you in the class</strong></p>

<pre><code class="r">setwd(&quot;/researchdata/fhgfs/hpc30/&quot;)
</code></pre>

<pre><code>## Error in setwd(&quot;/researchdata/fhgfs/hpc30/&quot;): cannot change working directory
</code></pre>

<pre><code class="r">inDir &lt;- getwd()#specify input directory
outDir &lt;- paste0(getwd(),&quot;/R_downstream&quot;) #specify output directory
phy &lt;- import_biom(BIOMfilename = paste0(inDir,&quot;/otus_table.tax.biom&quot;), 
        verbose = TRUE)#
</code></pre>

<pre><code>## Error in file(con, &quot;r&quot;): cannot open the connection
</code></pre>

<pre><code class="r">ntaxa(phy) #(number of OTUs)
</code></pre>

<pre><code>## Error in ntaxa(phy): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;ntaxa&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">sample_names(phy) &lt;- sub(&quot;\\/1&quot;,&quot;&quot;,sample_names(phy))#remove &quot;/1&quot; from filenames
</code></pre>

<pre><code>## Error in sample_names(phy): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;sample_names&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">#add phylogenetic tree (.tre file generated in QIIME)
tree &lt;- read_tree_greengenes(paste0(inDir,&quot;/otus_repsetOUT_aligned_pfiltered.tre&quot;))
</code></pre>

<pre><code>## Warning in file(con, &quot;r&quot;): cannot open file &#39;/home/kviljoen/16SrRNA-hex-
## tutorial/downstream/otus_repsetOUT_aligned_pfiltered.tre&#39;: No such file or
## directory
</code></pre>

<pre><code>## Error in file(con, &quot;r&quot;): cannot open the connection
</code></pre>

<pre><code class="r">#merge phy and tree
phy &lt;- merge_phyloseq(phy,tree)
</code></pre>

<pre><code>## Error in merge_phyloseq(phy, tree): object &#39;phy&#39; not found
</code></pre>

<p><strong>Data cleanup</strong></p>

<pre><code class="r">colnames(tax_table(phy))
</code></pre>

<pre><code>## Error in colnames(tax_table(phy)): error in evaluating the argument &#39;x&#39; in selecting a method for function &#39;colnames&#39;: Error in tax_table(phy) : 
##   error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">colnames(tax_table(phy)) &lt;-  c(&quot;Kingdom&quot;, &quot;Phylum&quot; , &quot;Class&quot; , &quot;Order&quot; , &quot;Family&quot; , &quot;Genus&quot;, &quot;Species&quot;)#e.g. replace &quot;Rank1&quot; with &quot;Kingdom&quot;
</code></pre>

<pre><code>## Error in colnames(tax_table(phy)) &lt;- c(&quot;Kingdom&quot;, &quot;Phylum&quot;, &quot;Class&quot;, &quot;Order&quot;, : object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">#clean taxonomic annotations, at the moment they are for example &#39;k__Bacteria&#39;; &#39;p_Firmicutes&#39; - remove k__ and p__ ...
tax_table(phy)[,&quot;Kingdom&quot;] &lt;- sub(&quot;k__&quot;,&quot;&quot;,tax_table(phy)[,&quot;Kingdom&quot;])
</code></pre>

<pre><code>## Error in tax_table(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">tax_table(phy)[,&quot;Phylum&quot;] &lt;- sub(&quot;p__&quot;,&quot;&quot;,tax_table(phy)[,&quot;Phylum&quot;])
</code></pre>

<pre><code>## Error in tax_table(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">tax_table(phy)[,&quot;Class&quot;] &lt;- sub(&quot;c__&quot;,&quot;&quot;,tax_table(phy)[,&quot;Class&quot;])
</code></pre>

<pre><code>## Error in tax_table(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">tax_table(phy)[,&quot;Order&quot;] &lt;- sub(&quot;o__&quot;,&quot;&quot;,tax_table(phy)[,&quot;Order&quot;])
</code></pre>

<pre><code>## Error in tax_table(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">tax_table(phy)[,&quot;Family&quot;] &lt;- sub(&quot;f__&quot;,&quot;&quot;,tax_table(phy)[,&quot;Family&quot;])
</code></pre>

<pre><code>## Error in tax_table(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">tax_table(phy)[,&quot;Genus&quot;] &lt;- sub(&quot;g__&quot;,&quot;&quot;,tax_table(phy)[,&quot;Genus&quot;])
</code></pre>

<pre><code>## Error in tax_table(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">tax_table(phy)[,&quot;Species&quot;] &lt;- sub(&quot;s__&quot;,&quot;&quot;,tax_table(phy)[,&quot;Species&quot;])
</code></pre>

<pre><code>## Error in tax_table(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">t= which(is.na(tax_table(phy)[,&quot;Phylum&quot;])) 
</code></pre>

<pre><code>## Error in tax_table(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">tax_table(phy) = tax_table(phy)[-t,] #remove rows that don&#39;t at least have Phylum-level annotation
</code></pre>

<pre><code>## Error in tax_table(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;tax_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<p><strong>Import metadata and merge with phyloseq object</strong></p>

<pre><code class="r">meta &lt;-  read.table(paste0(inDir,&quot;/practice.dataset1.metadata.tsv&quot;), sep = &quot;\t&quot;, header =TRUE, row.names=1)
</code></pre>

<pre><code>## Warning in file(file, &quot;rt&quot;): cannot open file &#39;/home/kviljoen/16SrRNA-
## hex-tutorial/downstream/practice.dataset1.metadata.tsv&#39;: No such file or
## directory
</code></pre>

<pre><code>## Error in file(file, &quot;rt&quot;): cannot open the connection
</code></pre>

<pre><code class="r">head(meta)
</code></pre>

<pre><code>## Error in head(meta): object &#39;meta&#39; not found
</code></pre>

<pre><code class="r">rownames(meta)
</code></pre>

<pre><code>## Error in rownames(meta): error in evaluating the argument &#39;x&#39; in selecting a method for function &#39;rownames&#39;: Error: object &#39;meta&#39; not found
</code></pre>

<pre><code class="r">head(sample_names(phy))
</code></pre>

<pre><code>## Error in sample_names(phy): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;sample_names&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">length(sample_names(phy))#15
</code></pre>

<pre><code>## Error in sample_names(phy): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;sample_names&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">length(rownames(meta))#15 (check if same number of samples in .biom file and metadatafile)
</code></pre>

<pre><code>## Error in rownames(meta): error in evaluating the argument &#39;x&#39; in selecting a method for function &#39;rownames&#39;: Error: object &#39;meta&#39; not found
</code></pre>

<pre><code class="r">length(intersect(rownames(meta),sample_names(phy)))#15 (check that the sample names match in all cases)
</code></pre>

<pre><code>## Error in intersect(rownames(meta), sample_names(phy)): error in evaluating the argument &#39;x&#39; in selecting a method for function &#39;intersect&#39;: Error in rownames(meta) : 
##   error in evaluating the argument &#39;x&#39; in selecting a method for function &#39;rownames&#39;: Error: object &#39;meta&#39; not found
</code></pre>

<pre><code class="r">sample_data(phy) &lt;- meta#assign the metadata to the phyloseq object &#39;phy&#39; (phyloseq will put these in the right order)
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;meta&#39; not found
</code></pre>

<pre><code class="r">nsamples(phy)
</code></pre>

<pre><code>## Error in nsamples(phy): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;nsamples&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">str(sample_data(phy))#need to change treatment column to factor variable
</code></pre>

<pre><code>## Error in sample_data(phy): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;sample_data&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">sample_data(phy)[,&quot;Treatment&quot;] &lt;- as.factor(unlist(sample_data(phy)[,&quot;Treatment&quot;]))
</code></pre>

<pre><code>## Error in unlist(sample_data(phy)[, &quot;Treatment&quot;]): error in evaluating the argument &#39;x&#39; in selecting a method for function &#39;unlist&#39;: Error in sample_data(phy) : 
##   error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;sample_data&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<p><strong>Save phyloseq object as an .RData file</strong></p>

<pre><code class="r">save(phy, file = paste0(outDir,&quot;/CBIO_16s_cert.RData&quot;)) #Save annotated object as a .RData object for quick reload if required at a later stage
</code></pre>

<pre><code>## Error in save(phy, file = paste0(outDir, &quot;/CBIO_16s_cert.RData&quot;)): object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">#load(paste0(outDir,&quot;/CBIO_16s_cert.RData&quot;)) #this is how you would reload the .RData object &#39;phy&#39;
</code></pre>

<h2>Explore number of reads per sample, make rarefaction curves and filter data as necessary</h2>

<p><strong>Explore number of reads per sample</strong></p>

<pre><code class="r">reads &lt;- sample_sums(phy)
</code></pre>

<pre><code>## Error in otu_table(x): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;otu_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">length(which(reads&lt;5000))
</code></pre>

<pre><code>## Error in which(reads &lt; 5000): object &#39;reads&#39; not found
</code></pre>

<pre><code class="r">raremax &lt;- min(reads)
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;reads&#39; not found
</code></pre>

<pre><code class="r">raremax
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;raremax&#39; not found
</code></pre>

<pre><code class="r">r=rarecurve(t(otu_table(phy)), step = 100, sample = raremax,xlab = &quot;number of reads/sample&quot;, ylab = &quot;number of OTUs&quot;,
        label = FALSE, xlim = c(0,100000))
</code></pre>

<pre><code>## Error in t(otu_table(phy)): error in evaluating the argument &#39;x&#39; in selecting a method for function &#39;t&#39;: Error in otu_table(phy) : 
##   error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;otu_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">pdf(paste0(outDir,&quot;/rarefaction_curve.pdf&quot;))
</code></pre>

<pre><code>## Error in pdf(paste0(outDir, &quot;/rarefaction_curve.pdf&quot;)): cannot open file &#39;/home/kviljoen/16SrRNA-hex-tutorial/downstream/R_downstream/rarefaction_curve.pdf&#39;
</code></pre>

<pre><code class="r">r
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;r&#39; not found
</code></pre>

<pre><code class="r">dev.off()
</code></pre>

<pre><code>## null device 
##           1
</code></pre>

<p>All samples have sufficient sequencing depth for inclusion in downstream analyses. The vertical line in the above plot indicates the sample with the lowest number of reads. Now we will scale data to account for differences in the number of reads/sample and filter rare OTUs that are not of biological interest for the purpose of this analysis (e.g. occurs only in one sample).</p>

<p><strong>Standardize abundances to median sequence depth</strong></p>

<pre><code class="r">total = median(sample_sums(phy))
</code></pre>

<pre><code>## Error in otu_table(x): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;otu_table&#39;: Error: object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">standf = function(x, t=total) round(t * (x / sum(x)))
M.std = transform_sample_counts(phy, standf)
</code></pre>

<pre><code>## Error in fun(1:10): object &#39;total&#39; not found
</code></pre>

<p><strong>Apply mild OTU filter</strong></p>

<p>Select OTUs where the rowsum for that OTU has at least 20% of samples with a count of 10 each OR where that OTU &gt; 0.001% of the total median count (for cases where the minority of samples may have high counts of a rare OTU)</p>

<pre><code class="r">M.f = filter_taxa(M.std,function(x) sum(x &gt; 10) &gt; (0.02*length(x)) | sum(x) &gt; 0.001*total, TRUE)
</code></pre>

<pre><code>## Error in inherits(x, get.component.classes()): object &#39;M.std&#39; not found
</code></pre>

<pre><code class="r">ntaxa(M.f)
</code></pre>

<pre><code>## Error in ntaxa(M.f): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;ntaxa&#39;: Error: object &#39;M.f&#39; not found
</code></pre>

<h2><strong>Basic exploratory plots: alpha- and beta-diversity, barplots, heatmap</strong></h2>

<p><strong>Alpha diversity by dog</strong></p>

<pre><code class="r">p &lt;- plot_richness(M.std,x = &quot;Dog&quot;,color = &quot;Treatment&quot;,measures=c(&quot;Shannon&quot;), 
        title = paste0(&quot;Standardized to total reads, N=&quot;,nsamples(M.std)))+theme(axis.text=element_text(size=16, face=&quot;bold&quot;),
                axis.title=element_text(size=16,face=&quot;bold&quot;))+geom_point(size=5)
</code></pre>

<pre><code>## Error in otu_table(physeq): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;otu_table&#39;: Error: object &#39;M.std&#39; not found
</code></pre>

<pre><code class="r">pdf(paste0(outDir,&quot;/alpha_diversity_by_dog_treatment.pdf&quot;))
</code></pre>

<pre><code>## Error in pdf(paste0(outDir, &quot;/alpha_diversity_by_dog_treatment.pdf&quot;)): cannot open file &#39;/home/kviljoen/16SrRNA-hex-tutorial/downstream/R_downstream/alpha_diversity_by_dog_treatment.pdf&#39;
</code></pre>

<pre><code class="r">p
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;p&#39; not found
</code></pre>

<pre><code class="r">dev.off()
</code></pre>

<pre><code>## null device 
##           1
</code></pre>

<p>Is there a significant difference in alpha diversity between dogs irrespective of treatment?</p>

<pre><code class="r">est &lt;- estimate_richness(M.f, split = TRUE, measures = c(&quot;Shannon&quot;))
</code></pre>

<pre><code>## Error in otu_table(physeq): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;otu_table&#39;: Error: object &#39;M.f&#39; not found
</code></pre>

<pre><code class="r">temp &lt;- cbind(est,sample_data(M.f)[,&quot;Dog&quot;])
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;est&#39; not found
</code></pre>

<pre><code class="r">head(temp)
</code></pre>

<pre><code>## Error in head(temp): object &#39;temp&#39; not found
</code></pre>

<pre><code class="r">t &lt;- kruskal.test(temp[,1]~temp[,2])
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;temp&#39; not found
</code></pre>

<pre><code class="r">t
</code></pre>

<pre><code>## standardGeneric for &quot;t&quot; defined from package &quot;base&quot;
## 
## function (x) 
## standardGeneric(&quot;t&quot;)
## &lt;environment: 0x5e29b18&gt;
## Methods may be defined for arguments: x
## Use  showMethods(&quot;t&quot;)  for currently available ones.
</code></pre>

<pre><code class="r">dunn.test(temp[,1],temp[,2])#post-hoc testing to see which dogs are different
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;dunn.test&quot;
</code></pre>

<p>Dog G has higher alpha diversity than dogs K and B irrespective of treatment, but this difference is not significant</p>

<p><strong>Alpha diversity by treatment</strong></p>

<pre><code class="r">p &lt;- plot_richness(M.std,x = &quot;Treatment&quot;,color = &quot;Dog&quot;,measures=c(&quot;Shannon&quot;), 
                title = paste0(&quot;Standardized to total reads, N=&quot;,nsamples(M.std)))+theme(axis.text=element_text(size=16, face=&quot;bold&quot;),
                axis.title=element_text(size=16,face=&quot;bold&quot;))+geom_point(size=5)
</code></pre>

<pre><code>## Error in otu_table(physeq): error in evaluating the argument &#39;object&#39; in selecting a method for function &#39;otu_table&#39;: Error: object &#39;M.std&#39; not found
</code></pre>

<pre><code class="r">pdf(paste0(outDir,&quot;/alpha_diversity_by_treatment_dog.pdf&quot;))
</code></pre>

<pre><code>## Error in pdf(paste0(outDir, &quot;/alpha_diversity_by_treatment_dog.pdf&quot;)): cannot open file &#39;/home/kviljoen/16SrRNA-hex-tutorial/downstream/R_downstream/alpha_diversity_by_treatment_dog.pdf&#39;
</code></pre>

<pre><code class="r">p
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;p&#39; not found
</code></pre>

<pre><code class="r">dev.off()
</code></pre>

<pre><code>## null device 
##           1
</code></pre>

<p>Are there significant differences in alpha diversity by treatment?</p>

<pre><code class="r">temp &lt;- cbind(est,sample_data(M.f)[,&quot;Treatment&quot;])
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;est&#39; not found
</code></pre>

<pre><code class="r">head(temp)
</code></pre>

<pre><code>## Error in head(temp): object &#39;temp&#39; not found
</code></pre>

<pre><code class="r">t &lt;- kruskal.test(temp[,1]~temp[,2])
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;temp&#39; not found
</code></pre>

<pre><code class="r">t
</code></pre>

<pre><code>## standardGeneric for &quot;t&quot; defined from package &quot;base&quot;
## 
## function (x) 
## standardGeneric(&quot;t&quot;)
## &lt;environment: 0x5e29b18&gt;
## Methods may be defined for arguments: x
## Use  showMethods(&quot;t&quot;)  for currently available ones.
</code></pre>

<pre><code class="r">dunn.test(temp[,1],temp[,2])
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;dunn.test&quot;
</code></pre>

<p><strong>Beta diversity using NMDS with Bray-Curtis as distance metric</strong></p>

<pre><code class="r">set.seed(2)
GP.ord.BC &lt;- ordinate(M.f, &quot;NMDS&quot;, &quot;bray&quot;, k=2, trymax=100)
</code></pre>

<pre><code>## Error in ordinate(M.f, &quot;NMDS&quot;, &quot;bray&quot;, k = 2, trymax = 100): object &#39;M.f&#39; not found
</code></pre>

<pre><code class="r">GP.ord.BC
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;GP.ord.BC&#39; not found
</code></pre>

<pre><code class="r">color = c(&quot;Treatment&quot;)
shape = c(&quot;Dog&quot;)
title=c(&quot;NMDS of 16S microbiome,Bray-Curtis distance,k=2&quot;)
MDS = plot_ordination(M.f, GP.ord.BC, color = color,shape=shape, 
        title = title)
</code></pre>

<pre><code>## Error in plot_ordination(M.f, GP.ord.BC, color = color, shape = shape, : object &#39;M.f&#39; not found
</code></pre>

<pre><code class="r">MDS.1  = MDS +theme(axis.text=element_text(size=16, face=&quot;bold&quot;),
                axis.title=element_text(size=18,face=&quot;bold&quot;), legend.title=element_text(size=14))+
        theme_bw()+labs(color=color, shape=shape)+geom_point(size=5)
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;MDS&#39; not found
</code></pre>

<pre><code class="r">pdf(paste0(outDir,&quot;/NMDS_Dogs_treatment_Bray_Curtis.pdf&quot;),8,5)
</code></pre>

<pre><code>## Error in pdf(paste0(outDir, &quot;/NMDS_Dogs_treatment_Bray_Curtis.pdf&quot;), 8, : cannot open file &#39;/home/kviljoen/16SrRNA-hex-tutorial/downstream/R_downstream/NMDS_Dogs_treatment_Bray_Curtis.pdf&#39;
</code></pre>

<pre><code class="r">MDS.1
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;MDS.1&#39; not found
</code></pre>

<pre><code class="r">dev.off()
</code></pre>

<pre><code>## null device 
##           1
</code></pre>

<p><strong>Beta diversity using NMDS with Unifrac as distance metric</strong></p>

<pre><code class="r">GP.ord.U &lt;- ordinate(M.f, &quot;NMDS&quot;, &quot;unifrac&quot;)
</code></pre>

<pre><code>## Error in ordinate(M.f, &quot;NMDS&quot;, &quot;unifrac&quot;): object &#39;M.f&#39; not found
</code></pre>

<pre><code class="r">GP.ord.U
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;GP.ord.U&#39; not found
</code></pre>

<pre><code class="r">color = c(&quot;Treatment&quot;)
shape = c(&quot;Dog&quot;)

title=c(&quot;NMDS of 16S microbiome, Unifrac distance, k=2&quot;)

MDS = plot_ordination(M.f, GP.ord.U, color = color, shape=shape, 
        title = title)
</code></pre>

<pre><code>## Error in plot_ordination(M.f, GP.ord.U, color = color, shape = shape, : object &#39;M.f&#39; not found
</code></pre>

<pre><code class="r">MDS.1  = MDS +theme(axis.text=element_text(size=16, face=&quot;bold&quot;),
                axis.title=element_text(size=18,face=&quot;bold&quot;), legend.title=element_text(size=14))+
        theme_bw()+labs(color=color)+geom_point(size=5)
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;MDS&#39; not found
</code></pre>

<pre><code class="r">pdf(paste0(outDir,&quot;/NMDS_Dogs_treatment_Unifrac.pdf&quot;),8,5)
</code></pre>

<pre><code>## Error in pdf(paste0(outDir, &quot;/NMDS_Dogs_treatment_Unifrac.pdf&quot;), 8, 5): cannot open file &#39;/home/kviljoen/16SrRNA-hex-tutorial/downstream/R_downstream/NMDS_Dogs_treatment_Unifrac.pdf&#39;
</code></pre>

<pre><code class="r">MDS.1
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;MDS.1&#39; not found
</code></pre>

<pre><code class="r">dev.off()
</code></pre>

<pre><code>## null device 
##           1
</code></pre>

<p><strong>Create a heatmap of taxa merged at the lowest available taxonomic level</strong></p>

<pre><code class="r">M.phy &lt;- tax_glom.kv(M.f)#this function is available in the &#39;microbiome_custom_functions.R&#39; script loaded at the beginning of this script
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;tax_glom.kv&quot;
</code></pre>

<pre><code class="r">ntaxa(M.phy)
</code></pre>

<pre><code>## Error in ntaxa(M.phy): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;ntaxa&#39;: Error: object &#39;M.phy&#39; not found
</code></pre>

<pre><code class="r">filename &lt;- c(&quot;cbio_cert_heatmap_merged_taxa&quot;)
main &lt;- paste(&quot;Merged taxa, Bray-Curtis distance&quot;)
f = paste0(outDir,&quot;/&quot;,filename,&quot;.pdf&quot;)
#color specification for column annotations above heatmap:
D.cols = c(&quot;B&quot;=&quot;#CC79A7&quot;,&quot;G&quot;=&quot;#56B4E9&quot;,&quot;K&quot;=&quot;#F0E442&quot;)
colours = list(Dog=D.cols)

#create distance matrix and calculate tree:
set.seed(2)
diss &lt;- distance(M.phy,method = &quot;bray&quot;, type = &quot;samples&quot;)
</code></pre>

<pre><code>## Error in distance(M.phy, method = &quot;bray&quot;, type = &quot;samples&quot;): object &#39;M.phy&#39; not found
</code></pre>

<pre><code class="r">clust.res&lt;-hclust(diss)
</code></pre>

<pre><code>## Error in hclust(diss): object &#39;diss&#39; not found
</code></pre>

<pre><code class="r">sample.order = clust.res$order
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;clust.res&#39; not found
</code></pre>

<pre><code class="r">#heatmap is output to file (the heatmap.k function can be found in the &#39;microbiome_custom_functions.R&#39; script)
hm = heatmap.k(physeq= M.phy,
        annot.cols = c(1,2),
        main = main,filename = f,colours=colours,Colv = sample.order,labrow = TRUE) 
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;heatmap.k&quot;
</code></pre>

<pre><code class="r">print(hm)
</code></pre>

<pre><code>## Error in print(hm): object &#39;hm&#39; not found
</code></pre>

<h2><strong>Barplots by dog</strong></h2>

<pre><code class="r">level = &quot;Genus&quot;
count = 500
perc = 0.25
#barplot will be written to file (the bar.plots function can be found in the &#39;microbiome_custom_functions.R&#39; script)
barplot = bar.plots(physeq = M.std,cat = &quot;Dog&quot;,level = level, count = count, perc = perc, outDir=outDir, 
        filen = &#39;Barplots_by_Dog&#39;)
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;bar.plots&quot;
</code></pre>

<pre><code class="r">print(barplot)
</code></pre>

<pre><code>## function (height, ...) 
## UseMethod(&quot;barplot&quot;)
## &lt;bytecode: 0xf933b8&gt;
## &lt;environment: namespace:graphics&gt;
</code></pre>

<h2>Detect taxa/OTUs that differ significantly by Dog</h2>

<p>convert phyloseq object to metagenomeSeq obj. NB use raw data not standardized:</p>

<pre><code class="r">Mraw.f = filter_taxa(phy,function(x) sum(x &gt; 10) &gt; (0.02*length(x)) | sum(x) &gt; 0.001*total, TRUE)
</code></pre>

<pre><code>## Error in inherits(x, get.component.classes()): object &#39;phy&#39; not found
</code></pre>

<pre><code class="r">ntaxa(Mraw.f)
</code></pre>

<pre><code>## Error in ntaxa(Mraw.f): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;ntaxa&#39;: Error: object &#39;Mraw.f&#39; not found
</code></pre>

<pre><code class="r">MGS=make_metagenomeSeq(Mraw.f)
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;make_metagenomeSeq&quot;
</code></pre>

<pre><code class="r">MGS
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;MGS&#39; not found
</code></pre>

<p><strong>Use Random forests analysis to detect taxa that are good predictors of Dog</strong></p>

<p>Example used: Dog G vs. Dog B (all treatment points)</p>

<pre><code class="r">sub.index &lt;- sample_names(M.f)[sample_data(M.f)[,&quot;Dog&quot;] != &quot;K&quot;]
</code></pre>

<pre><code>## Error in sample_names(M.f): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;sample_names&#39;: Error: object &#39;M.f&#39; not found
</code></pre>

<pre><code class="r">phy.temp &lt;- prune_samples(sub.index, M.f)
</code></pre>

<pre><code>## Error in prune_samples(sub.index, M.f): error in evaluating the argument &#39;samples&#39; in selecting a method for function &#39;prune_samples&#39;: Error: object &#39;sub.index&#39; not found
</code></pre>

<pre><code class="r">nsamples(phy.temp)
</code></pre>

<pre><code>## Error in nsamples(phy.temp): error in evaluating the argument &#39;physeq&#39; in selecting a method for function &#39;nsamples&#39;: Error: object &#39;phy.temp&#39; not found
</code></pre>

<pre><code class="r">RF.k(data = phy.temp, var = &quot;Dog&quot;, ntree=10000, cv.fold=10, outDir = outDir, Nfeatures.validation = 3)
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;RF.k&quot;
</code></pre>

<p>The class error rates are 0% (even one OTU enough to discriminate between Dog G and B?)</p>

<p>What if we used merged OTUs?</p>

<pre><code class="r">merged.phy &lt;- tax_glom.kv(phy.temp)
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;tax_glom.kv&quot;
</code></pre>

<pre><code class="r">RF.k(data = merged.phy, var = &quot;Dog&quot;, ntree=10000, cv.fold=10, outDir = outDir, Nfeatures.validation = 3, descriptor = &quot;merged_OTUs&quot;) #for details on RF.k() see microbiome_custom_functions.R file
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;RF.k&quot;
</code></pre>

<pre><code class="r">#Note that Nfeatures.validation: &#39;x&#39; number of top taxa to test (e.g. how good are the top 3 most important taxa at classifying)
</code></pre>

<p><strong>Differential abundance testing using MetagenomeSeq package</strong></p>

<p>Lets again compare dog G vs. dog B (merged taxa), this time using differential abundance testing</p>

<pre><code class="r">colours = list(Dog=D.cols)
a = super.fitZig.kv(physeq = merged.phy,factor = &quot;Dog&quot;,outDir = outDir,FileName =c(&quot;1_25FC_0.2_Dog_GvsB_taxa_merged&quot;),
        heatmap.descriptor=c(&quot;tax_annot&quot;), main=c(&quot;Dog G vs. B, taxa merged&quot;), subt=c(&quot;subt = FDR &lt; 0.05,|coeff| &gt;= 1.25, &gt;20%+ in either group&quot;), 
        ordered=TRUE, p=0.05, FC = 1.25, perc=0.2, extra.cols = c(&quot;Treatment&quot;))
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;super.fitZig.kv&quot;
</code></pre>

<pre><code class="r">print(a)
</code></pre>

<pre><code>## Error in print(a): object &#39;a&#39; not found
</code></pre>

<p>Now again compare dog G vs. dog B with differential abundance testing (individual OTUs as opposed to merged this time)</p>

<pre><code class="r">b = super.fitZig.kv(physeq = phy.temp,factor = &quot;Dog&quot;,outDir = outDir,FileName =c(&quot;1_25FC_0.2_Dog_GvsB_OTUs&quot;),
        heatmap.descriptor=c(&quot;tax_annot&quot;), main=c(&quot;Dog G vs. B, OTUs&quot;), subt=c(&quot;subt = FDR &lt; 0.05,|coeff| &gt;= 1.25, &gt;20%+ in either group&quot;), 
        ordered=TRUE, p=0.05, FC = 1.25, perc=0.2, extra.cols = c(&quot;Treatment&quot;))
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): could not find function &quot;super.fitZig.kv&quot;
</code></pre>

<pre><code class="r">b
</code></pre>

<pre><code>## Error in eval(expr, envir, enclos): object &#39;b&#39; not found
</code></pre>

<pre><code class="r">sessionInfo()
</code></pre>

<pre><code>## R version 3.0.2 (2013-09-25)
## Platform: x86_64-unknown-linux-gnu (64-bit)
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] NMF_0.20.5          Biobase_2.22.0      BiocGenerics_0.8.0 
##  [4] cluster_2.0.5       rngtools_1.2.4      pkgmaker_0.22      
##  [7] registry_0.2        randomForest_4.6-12 gridExtra_2.2.1    
## [10] ggplot2_1.0.1       phyloseq_1.6.1      picante_1.6-2      
## [13] nlme_3.1-131        vegan_2.4-2         lattice_0.20-34    
## [16] permute_0.9-4       ape_3.0-11          ade4_1.7-5         
## [19] knitr_1.11         
## 
## loaded via a namespace (and not attached):
##  [1] biom_0.3.12        Biostrings_2.30.1  codetools_0.2-15  
##  [4] colorspace_1.3-2   digest_0.6.12      doParallel_1.0.8  
##  [7] evaluate_0.10      foreach_1.4.3      formatR_1.4       
## [10] grid_3.0.2         gridBase_0.4-7     gtable_0.2.0      
## [13] igraph_1.0.1       IRanges_1.20.7     iterators_1.0.8   
## [16] magrittr_1.5       MASS_7.3-29        Matrix_1.2-8      
## [19] mgcv_1.8-17        multtest_2.18.0    munsell_0.4.3     
## [22] plyr_1.8.1         proto_1.0.0        RColorBrewer_1.0-5
## [25] Rcpp_0.12.9        reshape2_1.4.2     RJSONIO_1.3-0     
## [28] scales_0.4.1       splines_3.0.2      stats4_3.0.2      
## [31] stringi_1.1.2      stringr_1.2.0      survival_2.40-1   
## [34] tools_3.0.2        xtable_1.8-2       XVector_0.2.0
</code></pre>

</body>

</html>
